services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/GroupController.java

```diff
@@ -0,0 +1,119 @@
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeanUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+@Slf4j
+@RequestMapping("/groups")
+@RequiredArgsConstructor
+public class GroupController {
+
+  private final GroupRepository groupRepository;
+
+  private final UserGroupsRepository userGroupsRepo;
+
+  @PostMapping("")
+  public ResponseEntity<Object> createGroups(@Valid @RequestBody Group group) {
+    Group savedGroups = groupRepository.save(group);
+    IAuthUserWithPermissions currentUser =
+      (IAuthUserWithPermissions) SecurityContextHolder
+        .getContext()
+        .getAuthentication()
+        .getPrincipal();
+    UserGroup userGroup = UserGroup
+      .builder()
+      .group(new Group(savedGroups.getId()))
+      .userTenant(new UserTenant(currentUser.getUserTenantId()))
+      .isOwner(true)
+      .build();
+    userGroupsRepo.save(userGroup);
+    return new ResponseEntity<>(savedGroups, HttpStatus.CREATED);
+  }
+
+  @GetMapping("/count")
+  public ResponseEntity<Object> countGroups() {
+    Count count = Count.builder().totalCount(groupRepository.count()).build();
+    return new ResponseEntity<>(count, HttpStatus.OK);
+  }
+
+  @GetMapping("")
+  public ResponseEntity<Object> getAllGroups() {
+    List<Group> groupList = groupRepository.findAll();
+    return new ResponseEntity<>(groupList, HttpStatus.OK);
+  }
+
+  @GetMapping("{id}")
+  public ResponseEntity<Object> getGroupById(@PathVariable("id") UUID id) {
+    Optional<Group> group = groupRepository.findById(id);
+    if (group.isPresent()) {
+      return new ResponseEntity<>(group.get(), HttpStatus.OK);
+    } else {
+      throw new ResponseStatusException(
+        HttpStatus.NOT_FOUND,
+        "No group is present against given value"
+      );
+    }
+  }
+
+  @Transactional
+  @PatchMapping("{id}")
+  public ResponseEntity<Object> updateGroup(
+    @PathVariable("id") UUID id,
+    @RequestBody Group group
+  ) {
+    Group target;
+    Optional<Group> savedGroup = groupRepository.findById(id);
+    if (savedGroup.isPresent()) {
+      target = savedGroup.get();
+      BeanUtils.copyProperties(
```

Comment:
This is not validating the model right? what if i send some group property which should not be empty as a empty property.